hub/models.py

```diff
@@ -707,3 +724,514 @@ def cast_data(sender, instance, *args, **kwargs):
     elif instance.is_number and instance.int is None and instance.data:
         instance.int = int(instance.data)
         instance.data = ""
+
+from polymorphic.models import PolymorphicModel
+
+class ExternalDataSource(PolymorphicModel):
+    '''
+    A third-party data source that can be read and optionally written back to.
+    E.g. Google Sheet or an Action Network table.
+    This class is to be subclassed by specific data source types.
+    '''
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    # TODO: organisation = models.CharField(max_length=100)
```

Comment:
question: why is this commented out?

Reply:
Because I want it to be a foreign key to an actual Organisation object, but I don't have access to that in this branch.